### PR TITLE
add symlink during scaffolding

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR proposes adding a symlink from `vendor` to `non-vendor` packages during scaffolding. This, together with a symlink with `touch packages && ln -s vendor packages` in AEA dir will mean all import paths are detected.

Still to do:
- make sure local push is adjusted to still work

## Fixes

- #1923 (partially)

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: aea/helpers/sym_link.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+#
+# ------------------------------------------------------------------------------
+"""Sym link implementation for Linux, MacOS, and Windows."""
+
+import contextlib
+import os
+from functools import reduce
+from pathlib import Path
+
+
```
Comment (DavidMinarsch) [inline aea/helpers/sym_link.py:27]: straight copy from scripts

Comment (marcofavorito) [reply aea/helpers/sym_link.py:27]: OK

**Comment by pbukva** [comment]
@DavidMinarsch  I know that this is being used only for CI at the moment, so just comment: Simlinks are highly dependent on the File System and OS. The File System being the biggest potential hurdle here (since all relevant modern OSes with support symlinks).Issue w

File: aea/cli/fetch.py [hunk 2]
```diff
@@ -142,7 +150,7 @@ def _fetch_agent_deps(ctx: Context) -> None:
     :return: None
     :raises: ClickException re-raises if occurs in add_item call.
     """
-    for item_type in ("protocol", "contract", "connection", "skill"):
+    for item_type in (SKILL, PROTOCOL, CONNECTION, CONTRACT):
```
Comment (DavidMinarsch) [inline aea/cli/fetch.py:153]: fix

**Comment by DavidMinarsch** [comment]
> @DavidMinarsch I know that this is being used only for CI at the moment, so just comment: Simlinks are highly dependent on the File System and OS. The File System being the biggest potential hurdle here (since all modern and relevant OSes support symlinks at least for directories).
> Issue can pop specially on external disk drives which are usually formatted using FS with with NO symlink support (e.g. exFAT), and network drives (e.g. nfs, smb, etc ..., where FS server can be configured that way that creation symlinks via network api might be disabled, or not support at all). Also, Windows OS has `Create symbolic link` policy which needs to be enabled for groups/users in order to make creation of symlinks working ...

It's used for development related operations on CLI only. So these problems are unlikely to present themselves in this form.

**Pull request merged**